protocols/imap/src/main/java/org/apache/james/imap/api/ImapCommand.java

```diff
@@ -61,21 +68,7 @@ public String getName() {
     }
 
     public boolean validForState(ImapSessionState state) {
-        final boolean result;
-        switch (state) {
-        case AUTHENTICATED:
-            result = validInAuthenticated;
-            break;
-        case NON_AUTHENTICATED:
-            result = validInNonAuthenticated;
-            break;
-        case SELECTED:
-            result = validInSelected;
-            break;
-        default:
-            result = false;
-        }
-        return result;
+        return validity.validStates.contains(state);
```

Comment:
I think we can have a domain method name for this, like: `validity.allowed(state)`